File: src/BaseHmiTypes/Cycles/HmiCycle.cs
using BaseHmiTypes.Common;

namespace BaseHmiTypes.Cycles;

public sealed class HmiCycle : IHmiObject
{
    public DateTime? LastModified { get; set; }

    public string? Name { get; set; }

    public HmiMultilingualText? Comment { get; set; }

    public int ConsumerType { get; set; }

    public int CycleTime { get; set; }

    public string? CycleUnit { get; set; }

    public long FullCycleTime { get; set; }

    public int Number { get; set; }

    public bool StartAtStartingPoint { get; set; }

    public int StartingPointDay { get; set; }

    public int StartingPointHour { get; set; }

    public int StartingPointMinute { get; set; }

    public int StartingPointMonth { get; set; }

    public int StartingPointSecond { get; set; }

    public bool TriggerAtShutDown { get; set; }

    public bool TriggerAtStartUp { get; set; }
}

File: typescript/base-hmi-types/src/cycles/HmiCycle.ts
import { IHmiObject } from "../IHmiObject.js";
import { HmiMultilingualText } from "../common/HmiMultilingualText.js";

export class HmiCycle implements IHmiObject {
  lastModified?: Date;
  name?: string;
  comment?: HmiMultilingualText;
  consumerType = 0;
  cycleTime = 0;
  cycleUnit?: string;
  fullCycleTime = 0;
  number = 0;
  startAtStartingPoint = false;
  startingPointDay = 0;
  startingPointHour = 0;
  startingPointMinute = 0;
  startingPointMonth = 0;
  startingPointSecond = 0;
  triggerAtShutDown = false;
  triggerAtStartUp = false;
}
